oximeter/instruments/src/kstat/sampler.rs

```diff
@@ -1250,9 +1310,21 @@ impl oximeter::Producer for KstatSampler {
         loop {
             match rx.try_recv() {
                 Ok(sample) => samples.push(sample),
-                Err(mpsc::error::TryRecvError::Empty) => break,
-                Err(mpsc::error::TryRecvError::Disconnected) => {
-                    panic!("kstat stampler self-stat queue tx disconnected");
+                Err(broadcast::error::TryRecvError::Empty) => break,
+                Err(broadcast::error::TryRecvError::Lagged(n_missed)) => {
+                    warn!(
+                        self.log,
+                        "producer missed some samples because they're \
+                        being produced faster than `oximeter` is collecting";
+                        "n_missed" => %n_missed,
+                    );
+                }
+                Err(broadcast::error::TryRecvError::Closed) => {
+                    debug!(
+                        self.log,
+                        "kstat stampler self-stat queue tx disconnected"
+                    );
```

Comment:
do we expect this to ever happen during normal operation? should it be logged someplace?

Reply:
I'd only expect this during shutdown, or if the worker task panics. By "logged someplace", do you mean at a different level than `debug`?

Reply:
Ah, in that case, it's probably fine as is, carry on!